### ISSUE TITLE
Is deepflux reproducible

### ISSUE THREAD

**Issue by Tcorpion** [sunsmarterjie/SDL-Skeleton#8]
Follow the deepflux training, I could NOT get skeletons on par with “Deepflux for skeletons in the wild CVPR2019” reported. 
The loss always went to 'Nan' in a few steps in training  when LR=1e-6, after reducing [grad](https://github.com/sunsmarterjie/SDL-Skeleton/blob/main/modules/weighted_euclidean_loss.py) from ` grad = distL1 * (weightPos + weightNeg) / len(crop)` to ` grad = distL1 * (weightPos + weightNeg) / len(crop) / (weightPos + weightNeg).sum()` , here `(weightPos + weightNeg).sum()` was about 1e4, the loss finally converged.

In fact, the skeletons produced in Deepflux are really bad. For comparison, I trained same deepflux model with skeleton mask instead of flux field, the results were much better than that from flux field.

Did you reproduce the skeleton prediction as [paper reported](https://github.com/YukangWang/DeepFlux)?

**Comment by Tcorpion** [comment]
By the way, after replacing [upsampling](https://github.com/sunsmarterjie/SDL-Skeleton/blob/main/networks/deep_flux.py) to deconv2d as [Deepflux](https://github.com/YukangWang/DeepFlux) did, the loss could NOT converge with any values of learning rate.

**Comment by sunsmarterjie** [comment]
Thanks for your efforts !  Training deepflux needs Adam optimizer as my experience.  But I still did not reproduce the performance as the paper reported. Welcome to join the project and reproduce deepflux.

**Comment by Tcorpion** [comment]
I adopt adam optimizer and it help converge. The converged model of flux field is  NOT on par with skeleton mask training.
